jwave/geometry.py

```diff
@@ -15,10 +15,12 @@
 
 import math
 from dataclasses import dataclass
+from functools import partial
 from typing import List, Tuple, Union
+from numpy.typing import ArrayLike
 
 import numpy as np
-from jax import numpy as jnp
+from jax import numpy as jnp, custom_jvp
```

Comment:
I believe that the `custom_jvp` is not used anymore?

If you want, you can use the `pre-commit` of this repo to automatically take care of unused imports and other similarly annoying things :)
If you have never used it before (and you don't have to!), you can find here how to use it: https://github.com/ucl-bug/jwave/blob/main/CONTRIBUTING.md#getting-started